bundles/org.openhab.binding.bsblan/src/main/java/org/openhab/binding/bsblan/internal/handler/BsbLanBaseThingHandler.java

```diff
@@ -0,0 +1,128 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+
+import org.openhab.binding.bsblan.internal.configuration.BsbLanBridgeConfiguration;
+import org.openhab.binding.bsblan.internal.api.BsbLanApiCaller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Basic Handler class for all BSB-LAN services.
+ *
+ * @author Peter Schraffl - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BsbLanBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BsbLanBaseThingHandler.class);
+    private @Nullable BsbLanBridgeHandler bridgeHandler;
+
+    public BsbLanBaseThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            setChannel(channelUID.getId(), command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        BsbLanBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null) {
+            logger.debug("Initializing '{}': thing is only supported within a bridge", getDescription());
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+        logger.debug("Initializing '{}' thing", getDescription());
+        bridgeHandler.registerThing(this);
+    }
+
+    protected synchronized @Nullable BsbLanBridgeHandler getBridgeHandler() {
+        if (this.bridgeHandler == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                return null;
+            }
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof BsbLanBridgeHandler) {
+                this.bridgeHandler = (BsbLanBridgeHandler)handler;
+            }
+        }
+        return this.bridgeHandler;
+    }
+
+    /**
+     * Update all channels
+     */
+    protected void updateChannels() {
+        for (Channel channel : getThing().getChannels()) {
+            updateChannel(channel.getUID().getId());
+        }
+    }
+
+    protected @Nullable BsbLanApiCaller getApiCaller() {
+        // use a local variable to avoid the build warning "Potential null pointer access"
+        BsbLanBridgeHandler localBridgeHandler = bridgeHandler;
+        if (localBridgeHandler == null)
+            return null;
```

Comment:
Openhab's formatting standard requires brackets for all if statements.

Reply:
fixed